docs/sdks/uid2-sdk-ref-java.md

```diff
@@ -102,6 +103,183 @@ Available information returned through the SDK is outlined in the following tabl
+   if (identity != null) { String advertisingToken = identity.getAdvertisingToken(); }
+   ```
+3. When the user accesses another page, or on a timer, determine whether a refresh is needed:
+   1. Retrieve the identity JSON string from the user's session, and then call the following function that takes the identity information as input and generates an `IdentityTokens` object:
+
+      ```java
+      IdentityTokens identity = IdentityTokens.fromJsonString(identityJsonString);
+      ```
+   2. Determine if the identity can be refreshed (that is, the refresh token hasn't expired):
+
+      ```java
+      if (identity == null || !identity.isRefreshable()) { we must no longer use this identity (for example, remove this identity from the user's session) }
+      ```
+   3. Determine if a refresh is needed:
+
+      ```java
+      if (identity.isDueForRefresh()) {..}
+      ```
+4. If needed, refresh the token and associated values:
+ 
+   ```java
+   TokenRefreshResponse tokenRefreshResponse = publisherUid2Client.refreshToken(identity);
+   ```
+ 
+5. Store `tokenRefreshResponse.getIdentityJsonString()` in the user's session.
+
+   If the user has opted out, this method returns `null`, indicating that the user's identity should be removed from the session. To confirm optout, you can use the `tokenRefreshResponse.isOptout()` function.
+
+### Advanced Usage
+
+1. Create an instance of `PublisherUid2Helper` as an instance variable:
+
+    ```java
+    private final PublisherUid2Helper publisherUid2Helper = new PublisherUid2Helper(UID2_SECRET_KEY);
+    ```
+2. Call a function that takes the user's email address or phone number as input and creates a secure request data envelope. See [Encrypting requests](../getting-started/gs-encryption-decryption.md#encrypting-requests). The following example uses an email address:
+
+    ```java
+    EnvelopeV2 envelope = publisherUid2Helper.createEnvelopeForTokenGenerateRequest(TokenGenerateInput.fromEmail(emailAddress).doNotGenerateTokensForOptedOut());
+    ```
+3. Using an HTTP client library of your choice, post this envelope to the [POST token/generate](../endpoints/post-token-generate.md) endpoint, including headers and body:
+   1. Headers: Depending on your HTTP library, this might look something like the following:  
+    
+      `.putHeader("Authorization", "Bearer " + UID2_API_KEY)`  
+      `.putHeader("X-UID2-Client-Version", PublisherUid2Helper.getVersionHeader())`
+   2. Body: `envelope.getEnvelope()`
+   :::important
+   - Be sure to call this endpoint only when you have obtained legal basis to convert the user’s [directly identifying information (DII)](../ref-info/glossary-uid.md#gl-dii) to UID2 tokens for targeted advertising.
+
+   - Always apply `doNotGenerateTokensForOptedOut()`. This applies `policy=1` in the [/token/generate](../endpoints/post-token-generate.md#token-generation-policy) call. Support for `policy=0` will be removed soon. (**GWH_SW03 same comment as earlier occurrence.**)
```

Comment:
Again just remove
_Support for `policy=0` will be removed soon._

Reply:
Addressed same as your earlier comment, see query, thx.